sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -62,6 +77,19 @@
  * PCollection<String> lines = p.apply(TextIO.read().from("/local/path/to/file.txt"));
  * }</pre>
  *
+ * <p>Example 2: reading a PCollection of filenames.
+ *
+ * <pre>{@code
+ * Pipeline p = ...;
+ *
+ * // E.g. the filenames might be computed from other data in the pipeline, or
+ * // read from a data source.
+ * PCollection<String> filenames = ...;
+ *
+ * // Read all files in the collection.
+ * PCollection<String> lines = filenames.apply(TextIO.readAll());
```

Comment:
I'm still unhappy with readAll (it feels to me like it means read all lines), but I still don't have a better suggestion.